### PR TITLE
Dagger setup done, Also first attempt to an analytics manager

### PR DESCRIPTION
**Pull request by davidehcastaneda**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -33,6 +35,19 @@ android {
...
+    implementation 'com.google.firebase:firebase-analytics-ktx'
+    implementation 'com.google.firebase:firebase-database'
+    implementation 'com.google.firebase:firebase-storage'
+    implementation 'com.google.firebase:firebase-config'
+
+    implementation 'com.google.dagger:dagger:2.41'
+    implementation 'com.google.dagger:dagger-android:2.41'
+    implementation 'com.google.dagger:dagger-android-support:2.41'
+    kapt 'com.google.dagger:dagger-compiler:2.41'
+    kapt 'com.google.dagger:dagger-android-processor:2.41'
```
Comment (HEHCDraft) [inline app/build.gradle:49]: Why not manage this from a deps.gradle file?

File: app/src/main/java/com/theMexicoProject/theweatherapp/analytics/AnalyticsImpl.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class AnalyticsImpl : AnalyticsManager {
+
+    private val firebaseAnalytics: FirebaseAnalytics by lazy {
+        Firebase.analytics
+    }
+
+    override fun report(tag: String, event: Parcelable?) {
+        val bundle = Bundle()
+        bundle.putParcelable(FirebaseAnalytics.Param.ITEM_ID, event)
```
Comment (HEHCDraft) [inline app/src/main/java/com/theMexicoProject/theweatherapp/analytics/AnalyticsImpl.kt:17]: Why have a val created each time the function is called? Would it not be more practical to have a single call?

**Review by HEHCDraft — APPROVED** [review]
LGTM, few suggestions.

File: app/src/main/java/com/theMexicoProject/theweatherapp/analytics/AnalyticsImpl.kt (hunk 2 above)
Comment (davidehcastaneda) [reply app/src/main/java/com/theMexicoProject/theweatherapp/analytics/AnalyticsImpl.kt:17]: yeah!, ty. I just didnt like the way It looked lol

File: app/build.gradle (hunk 1 above)
Comment (davidehcastaneda) [reply app/build.gradle:49]: Yeah, Im probably adding that on the next PR

**Pull request merged**